lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp

```diff
@@ -61,6 +61,26 @@ std::vector<c10::ScalarType> compute_dtype_embedding_dense_backward(const at::Te
   return {grad_output.scalar_type()};
 }
 
+std::vector<std::vector<int64_t>> compute_shape_index_select(const at::Tensor & self, int64_t dim,
+    const at::Tensor & index) {
+  // Based on definition of https://pytorch.org/docs/stable/generated/torch.index_select.html.
+  // Promote Rank 0 index tensor to a 1 * 1 tensor.
+  auto index_dim = index.dim() > 0 ? index.dim() : 1;
+  auto index_size = index.dim() > 0 ? index.size(0) : 1;
+  TORCH_CHECK(index_dim == 1);
+  TORCH_CHECK(dim >= 0 && dim < self.dim());
```

Comment:
isn't this check wrong?

I think we should support negative dim's.

```
>>> torch.index_select(x, 0, torch.tensor([1,2]))
tensor([[0., 1., 0., 0., 0.],
        [0., 0., 1., 0., 0.]])
>>> torch.index_select(x, 1, torch.tensor([1,2]))
tensor([[0., 0.],
        [1., 0.],
        [0., 1.],
        [0., 0.],
        [0., 0.]])
>>> torch.index_select(x, -1, torch.tensor([1,2]))
tensor([[0., 0.],
        [1., 0.],
        [0., 1.],
        [0., 0.],
        [0., 0.]])
```

See how I handled this in compute_shape_cat:

```
  int64_t rank = tensors[0].sizes().size();
  int64_t canonical_dim = dim % rank;
  if (canonical_dim < 0) {
      canonical_dim += rank;
  }
  ```

Reply:
you could make `canonical_dim` a helper func at the top of LazyShapeDtype.cpp, i think

Reply:
Oh, negative dim. Let me adjust that.